modules/percolator/src/main/java/org/elasticsearch/percolator/PercolateQueryBuilder.java

```diff
@@ -517,9 +518,9 @@ protected Analyzer getWrappedAnalyzer(String fieldName) {
         final IndexSearcher docSearcher;
         final boolean excludeNestedDocuments;
         if (docs.size() > 1 || docs.get(0).docs().size() > 1) {
-            assert docs.size() != 1 || context.hasNested();
+            assert docs.size() != 1 || context.nestedLookup() != NestedLookup.EMPTY;
```

Comment:
I am tempted to ask you to leave hasNested given we call in quite some places and we could have this comparison within that method instead.

Reply:
Yeah I went back and forth on this a lot. In the end it seemed neater to me to have just one method, and the EMPTY implementation simplifies things at several callsites. But I'm not totally wedded to it.

Reply:
ok so it seems that the EMPTY implementation is used also for cases where NestedLookup is carried around despite hasNested would return false (if we still had it)? I am also torn, but I have a slight preference to leave hasNested rather than having that comparison with EMPTY in those places. Maybe EMPTY could even become private then.

Reply:
So I tried this out and I think I prefer it as it is - the vast majority of places this is replacing a simple `getNestedLookup()` with an `if(context.hasNested()) { NestedLookup nl = context.nestedLookup(); ... }`

Reply:
yea I see, that part I agree we should leave as-is. I was wondering if hasNested would be better solely as a replacement for the != EMPTY comparison, and maybe so that we can make EMPTY private. I haven't tried it though so you may tell me this won't work either.

Reply:
EMPTY is a static constant on an interface so we can't make it private, and it's used in another static method on that interface so it needs to be there really.  Which means that `hasNested()` is exactly equivalent to `nestedLookup() == NestedLookup.EMPTY` always.  I don't think it's worth bringing it back if we don't get any encapsulation benefits from it?

Reply:
I see, thanks for giving it some thought.